src/translator/beam_search.h

```diff
@@ -373,8 +373,11 @@ class BeamSearch {
                      first,
                      batch);
 
-      if (triePrune_) {
-         beams = filterForContinuations(beams, paraphrase_);
+      if (!first || !paraphrase_){ // if first word and paraphrasing, allow words in trie
```

Comment:
The comment says "and"  but you' have an or? 

Also you have an inner if statement `if (triePrune_)` that has no else. You should move it to the parent.

Reply:
Comment is the double negation version of the if-statement. I thought using `or` is faster..

Sure I changed that for readability.

Reply:
Suuuuure, but the branch predictor would learn those anyways. Also it's not a performance critical path.